### PR TITLE
feat(workspace): Support the OTP auxiliary authentication

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:
 workspace support the OTP auxiliary authentication.
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccService_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccService_basic -timeout 360m -parallel 4
=== RUN   TestAccService_basic
=== PAUSE TestAccService_basic
=== CONT  TestAccService_basic
--- PASS: TestAccService_basic (642.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 643.004s

$ make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccService_localAD'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccService_localAD -timeout 360m -parallel 4
=== RUN   TestAccService_localAD
=== PAUSE TestAccService_localAD
=== CONT  TestAccService_localAD
--- PASS: TestAccService_localAD (599.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 599.994s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go [hunk 1]
```diff
@@ -211,6 +255,13 @@ resource "huaweicloud_workspace_service" "test" {
 
   internet_access_port = 9001
   enterprise_id        = "%[2]s"
+
+  otp_config_info {
+    enable       = true
+    receive_mode = "VMFA"
+   	rule_type 	 = "ACCESS_MODE"
+	rule         = "PRIVATE"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go]:
```suggestion
    rule_type = "ACCESS_MODE"
    rule      = "PRIVATE"
```
invalid format

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go [hunk 2]
```diff
@@ -292,6 +343,54 @@ resource "huaweicloud_workspace_service" "test" {
...
+  ]
+
+  internet_access_port = 9001
+  enterprise_id        = "%[7]s"
+
+  otp_config_info {
+    enable       = true
+    receive_mode = "VMFA"
+   	rule_type 	 = "ACCESS_MODE"
+	rule         = "PRIVATE"
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go]: ditto

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go [hunk 3]
```diff
@@ -478,6 +560,41 @@ func updateServiceEnterpriseId(ctx context.Context, client *golangsdk.ServiceCli
...
+	opts := services.UpdateAuthConfigOpts{
+		AuthType:      "OTP",
+		OptConfigInfo: buildAuthConfig(authConfig[0].(map[string]interface{})),
+	}
+	err := services.UpdateAssistAuthConfig(client, opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go]:
```suggestion
func updateAssistAuthConfig(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
	authConfig := d.Get("otp_config_info").([]interface{})
	if len(authConfig) < 1 {
		return nil
	}

	opts := services.UpdateAuthConfigOpts{
		AuthType:      "OTP",
		OptConfigInfo: buildAuthConfig(authConfig[0].(map[string]interface{})),
	}
	return services.UpdateAssistAuthConfig(client, opts)
}
```
Both function updateAssistAuthConfig and function UpdateAssistAuthConfig are return the result of error type, there is unnecessary to check whether function error response is not nil.

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go [hunk 4]
```diff
@@ -199,6 +242,11 @@ func ResourceService() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"otp_config_info": {
+				Type:     schema.TypeList,
+				Optional: true,
```
Comment (Lance52259) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go:247]: maybe support a limit: `MaxItem: 1`

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go (hunk 1 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go]: fixed

File: huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/workspace/resource_huaweicloud_workspace_service_test.go]: fixed

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go (hunk 3 above)
Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go]: fixed

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go (hunk 4 above)
Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go:247]: fixed

**Comment by wuzhuanhong** [comment]
$ make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccService_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccService_basic -timeout 360m -parallel 4
=== RUN   TestAccService_basic
=== PAUSE TestAccService_basic
=== CONT  TestAccService_basic
--- PASS: TestAccService_basic (629.53s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 629.579s

make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccService_localAD'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccService_localAD -timeout 360m -parallel 4
=== RUN   TestAccService_localAD
=== PAUSE TestAccService_localAD
=== CONT  TestAccService_localAD
--- PASS: TestAccService_localAD (592.43s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 592.506s

File: huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go [hunk 5]
```diff
@@ -95,6 +95,49 @@ func securityGroupSchemaResource() *schema.Resource {
 	}
 }
 
+func AssistAuthSchemaResource() *schema.Resource {
```
Comment (ShiChangkuo) [inline huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go]: please use the lowercase if we do not export this function.
```suggestion
func assistAuthSchemaResource() *schema.Resource {
```

Comment (wuzhuanhong) [reply huaweicloud/services/workspace/resource_huaweicloud_workspace_service.go]: fixed

**Pull request merged**